Web.Client/Pages/Electives/StudentSubjectRegistrationProgressList.razor.cs

```diff
@@ -0,0 +1,51 @@
+﻿using MensaGymnazium.IntranetGen3.Contracts;
+using MensaGymnazium.IntranetGen3.Web.Client.Services.DataStores;
+
+namespace MensaGymnazium.IntranetGen3.Web.Client.Pages.Electives;
+
+public partial class StudentSubjectRegistrationProgressList
+{
+	[Inject] protected IStudentsDataStore StudentsDataStore { get; set; }
+	[Inject] protected IGradesDataStore GradesDataStore { get; set; }
+	[Inject] protected ISubjectRegistrationProgressValidationFacade ProgressValidationFacade { get; set; }
+
+	private HxGrid<StudentSubjectRegistrationProgressListItemDto> _grid; // @ref
+	private StudentSubjectRegistrationProgressListFilter _filterModel = new();
+
+	protected override async Task OnInitializedAsync()
+	{
+		await StudentsDataStore.EnsureDataAsync();
+		await GradesDataStore.EnsureDataAsync();
+	}
+
+	private async Task<GridDataProviderResult<StudentSubjectRegistrationProgressListItemDto>> DataProvider(
+		GridDataProviderRequest<StudentSubjectRegistrationProgressListItemDto> request)
+	{
+		var data = await ProgressValidationFacade
+			.GetProgressListAsync(_filterModel, request.CancellationToken);
+
+		// Due to serialization, sometimes we receive empty lists,
+		// so 'List<>MissingCriteria' may be null
+
```

Comment:
To, že je to kvůli serializaci je spíš takový guess :-D, protože server v odpovědi posílá vždy list, ale občas s velikostí 0. Serializuje se velikost 0 na null, nebo je tam jiný důvod?

Reply:
protobuf serializace nerozlišuje prázdnou kolekci od `null`. Na DTOs je tak potřeba mít `List`-membery předinicializované na `new()`. Pokud je `List` samotným DTO, pak je potřeba se s tím vypořádat ručně, třeba `List<T> result = await XyFacade.DoSomethingAsync() ?? new()`.